java-checks/src/test/files/checks/security/SecureXmlTransformerCheck.java

```diff
@@ -0,0 +1,88 @@
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.XMLConstants;
+
+class A {
+
+  TransformerFactory classField = TransformerFactory.newInstance();
+
+  TransformerFactory no_call_to_securing_method() {
+    TransformerFactory factory = TransformerFactory.newInstance(); // Noncompliant
+    return factory;
+  }
+
+  // setFeature
+
+  TransformerFactory secure_processing_true() {
+    TransformerFactory factory = TransformerFactory.newInstance();
+    factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+    return factory;
+  }
+
+  TransformerFactory secure_processing_false() {
+    TransformerFactory factory = TransformerFactory.newInstance(); // Noncompliant
+    factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, false);
+    return factory;
+  }
+
+  TransformerFactory secure_processing_true_with_literal() {
+    TransformerFactory factory = TransformerFactory.newInstance();
+    factory.setFeature("http://javax.xml.XMLConstants/feature/secure-processing", true);
+    return factory;
+  }
+
+  TransformerFactory secure_processing_true() {
```

Comment:
`setattribute_dtd_only` - actually this duplicates line 63 ... no, this is setFeature, so should be removed?

Reply:
Why should it be removed? I'm going to change the name of the method which is not helpful at all.